cloudevents/formats/cbor-format.md

```diff
@@ -0,0 +1,199 @@
+
+All REQUIRED and all not omitted OPTIONAL attributes in the given event MUST
+become members of the CBOR data item, with the respective CBOR item member name
+matching the attribute name, and the member's type and value being mapped using
+the [type system mapping](#22-type-system-mapping).
+
+OPTIONAL not omitted attributes MAY be represented as a `null` CBOR simple value.
+
+### 3.1. Handling of "data"
+
+The CBOR representation of the event "data" payload is determined by the runtime
+type of the `data` content and the value of the [`datacontenttype`
+attribute][datacontenttype].
+
+#### 3.1.1. Payload Encoding
+
+Before taking action, a [CBOR encoder][cbor-encoder] MUST first determine the runtime data
+type of the `data` content.
+
+If the implementation determines that the type of data is `Binary`, the value
+MUST be represented as a [Major type 2][cbor-major-types] value. If present, the `datacontenttype` MUST
+reflect the format of the original binary data.
+
+If the type of data is not `Binary`, the implementation will next determine
+whether the value of the `datacontenttype` attribute declares the `data` to
+contain CBOR-formatted content. Such a content type is defined as one having a
+[media subtype][rfc2045-sec5] equal to `cbor` or ending with a `+cbor` format
+extension. That is, a `datacontenttype` declares CBOR-formatted content if its
+media type, when stripped of parameters, has the form `*/cbor` or `*/*+cbor`.
+If the `datacontenttype` is unspecified, processing SHOULD proceed as if the
+`datacontenttype` had been specified explicitly as `application/cbor`.
+
+If the `datacontenttype` declares the data to contain CBOR-formatted content, a
+[CBOR encoder][cbor-encoder] MUST translate the data value to a [CBOR date item][cbor-data-item], and
+use the member name `data` to store it inside the CBOR representation. The data
+value MUST be stored directly as a CBOR data item, rather than as an encoded CBOR
+buffer (tagged or un-tagged). An implementation MAY fail to encode the
+event if it is unable to translate the runtime value to a [CBOR data item][cbor-data-item].
+
+Otherwise, if the `datacontenttype` does not declare CBOR-formatted data
+content, a [CBOR encoder][cbor-encoder] MUST store the data value, properly encoded according 
+to the `datacontenttype`, in the `data` member of the CBOR representation. 
+An implementation MAY fail to encode the event if it is unable to represent
+the runtime value as a properly encoded CBOR data item.
+
+Furthermore, unlike attributes, for which value types are restricted by the
+[type-system mapping](#22-type-system-mapping), the `data` member
+[CBOR data item][cbor-data-item] is unrestricted, and MAY contain any valid CBOR if the
+`datacontenttype` declares the data to be CBOR-formatted. In particular, the
+`data` member MAY have a simple value of `null`, representing an explicit `null`
+payload as distinct from the absence of the `data` member.
```

Comment:
So data=null is not the same as "missing data" ?

Reply:
yes, the same as the JSON format